lib/ansible/plugins/action/reboot.py

```diff
@@ -142,10 +143,15 @@ def get_shutdown_command(self, task_vars, distribution):
         if not isinstance(search_paths, list):
             raise AnsibleError("'search_paths' must be a string or list, got {0}: {1}".format(type(search_paths), search_paths))
 
-        # If we did get a list, ensure it's a flast list and does not contain nested data structures
+        # If we did get a list, ensure it's a flat list and does not contain nested data structures
         nested_list = any(isinstance(x, (dict, list)) for x in search_paths)
```

Comment:
redundant with the is_string check below now; just meld those two together (so you're keeping the more declarative "any" rather than the explicit loop) and you're good to go

Reply:
Thanks for the info. I simplified it quite a bit.